applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -549,125 +551,124 @@ static bool key_value_set(struct items *items, u16_t usage_id, s16_t value)
 			}
+		} else {
+			break;
 		}
+	}
 
-		/* Fill the rest of report with zeros. */
-		for (; cnt < ARRAY_SIZE(event->keys); cnt++) {
-			event->keys[cnt] = 0;
-		}
+	/* Fill the rest of report with zeros. */
+	for (; cnt < ARRAY_SIZE(event->keys); cnt++) {
+		event->keys[cnt] = 0;
+	}
 
-		EVENT_SUBMIT(event);
+	EVENT_SUBMIT(event);
 
-		rd->items.update_needed = false;
-	} else {
-		/* Not supported. */
-		__ASSERT_NO_MSG(false);
-	}
+	rd->items.update_needed = false;
 }
 
 static void send_report_mouse(void)
 {
-	if (IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_MOUSE_SUPPORT)) {
-		struct report_data *rd = &state.report_data[IN_REPORT_MOUSE];
+	if (!IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_MOUSE_SUPPORT)) {
+		/* Not supported. */
+		__ASSERT_NO_MSG(false);
+		return;
+	}
 
-		struct hid_mouse_event *event = new_hid_mouse_event();
+	struct report_data *rd = &state.report_data[IN_REPORT_MOUSE];
 
-		event->subscriber = state.selected->id;
+	struct hid_mouse_event *event = new_hid_mouse_event();
 
-		event->dx        = state.last_dx;
-		event->dy        = -state.last_dy;
-		event->wheel     = state.wheel_acc / 2;
-		event->button_bm = 0;
+	event->subscriber = state.selected->id;
 
-		state.last_dx   = 0;
-		state.last_dy   = 0;
-		state.wheel_acc -= event->wheel * 2;
+	event->dx        = state.last_dx;
+	event->dy        = -state.last_dy;
+	event->wheel     = state.wheel_acc / 2;
+	event->button_bm = 0;
 
-		/* Traverse pressed keys and build mouse buttons report */
-		for (size_t i = 0; i < ARRAY_SIZE(rd->items.item); i++) {
-			struct item item = rd->items.item[i];
+	state.last_dx   = 0;
+	state.last_dy   = 0;
+	state.wheel_acc -= event->wheel * 2;
 
-			if (item.usage_id) {
-				__ASSERT_NO_MSG(item.usage_id <= 8);
-				__ASSERT_NO_MSG(item.value > 0);
+	/* Traverse pressed keys and build mouse buttons report */
+	for (size_t i = 0; i < ARRAY_SIZE(rd->items.item); i++) {
+		struct item item = rd->items.item[i];
 
-				u8_t mask = 1 << (item.usage_id - 1);
+		if (item.usage_id) {
+			__ASSERT_NO_MSG(item.usage_id <= 8);
+			__ASSERT_NO_MSG(item.value > 0);
 
-				event->button_bm |= mask;
-			}
+			u8_t mask = 1 << (item.usage_id - 1);
+
+			event->button_bm |= mask;
 		}
+	}
 
-		EVENT_SUBMIT(event);
+	EVENT_SUBMIT(event);
 
-		rd->items.update_needed = false;
-	} else {
-		/* Not supported. */
-		__ASSERT_NO_MSG(false);
-	}
+	rd->items.update_needed = false;
 }
 
 static void send_report_ctrl(enum in_report report_type)
 {
-	if (IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_SYSTEM_CTRL_SUPPORT) ||
-	    IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_CONSUMER_CTRL_SUPPORT)) {
-		if (report_type == IN_REPORT_SYSTEM_CTRL) {
-			__ASSERT_NO_MSG(IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_SYSTEM_CTRL_SUPPORT));
-		} else if (report_type == IN_REPORT_CONSUMER_CTRL) {
-			__ASSERT_NO_MSG(IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_CONSUMER_CTRL_SUPPORT));
-		} else {
-			/* Unsupported report type. */
-			__ASSERT_NO_MSG(false);
-		}
+	if ((report_type == IN_REPORT_SYSTEM_CTRL) &&
```

Comment:
IS_ENABLED could be before report type check

Reply:
This will go away in the subsequent commit anyway...